_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -913,6 +925,112 @@ public function update_data( $request ) {
+	 *
+	 * @return string Result of onboarding processing and, if there is one, an error message.
+	 */
+	private function _process_onboarding( $data ) {
+		$error = array();
+
+		if ( ! empty( $data['siteTitle'] ) ) {
+			// If option value was the same, consider it done.
+			if ( ! ( update_option( 'blogname', $data['siteTitle'] ) || get_option( 'blogname' ) == $data['siteTitle'] ) ) {
+				$error[] = 'siteTitle';
+			}
+		}
+
+		if ( ! empty( $data['siteDescription'] ) ) {
+			// If option value was the same, consider it done.
+			if ( ! ( update_option( 'blogdescription', $data['siteDescription'] ) || get_option( 'blogdescription' ) == $data['siteDescription'] ) ) {
+				$error[] = 'siteDescription';
+			}
+		}
+
+		$site_title = get_option( 'blogname' );
+		$author = get_current_user_id() || 1;
+
+		// If $data['homepageFormat'] is 'posts', we have nothing to do since it's WordPress' default
+		if ( 'page' === $data['homepageFormat'] ) {
+			if ( ! ( update_option( 'show_on_front', 'page' ) || get_option( 'show_on_front' ) == 'page' ) ) {
+				$error[] = 'homepageFormat';
+			}
+
+			$home = wp_insert_post( array(
+				'post_type'     => 'page',
+				/* translators: this references the home page of a site, also called front page. */
+				'post_title'    => esc_html_x( 'Home Page', 'The home page of a website.', 'jetpack' ),
+				'post_content'  => sprintf( esc_html__( 'Welcome to %s.', 'jetpack' ), $site_title ),
+				'post_status'   => 'publish',
+				'post_author'   => $author,
+			) );
+			if ( 0 == $home ) {
+				$error[] = 'home insert: 0';
+			} elseif ( is_wp_error( $home ) ) {
+				$error[] = 'home creation: '. $home->get_error_message();
+			}
+			if ( ! ( update_option( 'page_on_front', $home ) || get_option( 'page_on_front' ) == $home ) ) {
+				$error[] = 'home set';
+			}
+
+			$blog = wp_insert_post( array(
+				'post_type'     => 'page',
+				/* translators: this references the page where blog posts are listed. */
+				'post_title'    => esc_html_x( 'Blog', 'The blog of a website.', 'jetpack' ),
+				'post_content'  => sprintf( esc_html__( 'These are the latest posts in %s.', 'jetpack' ), $site_title ),
+				'post_status'   => 'publish',
+				'post_author'   => $author,
+			) );
+			if ( 0 == $blog ) {
+				$error[] = 'blog insert: 0';
+			} elseif ( is_wp_error( $blog ) ) {
+				$error[] = 'blog creation: '. $blog->get_error_message();
+			}
+			if ( ! ( update_option( 'page_for_posts', $blog ) || get_option( 'page_for_posts' ) == $blog ) ) {
+				$error[] = 'blog set';
+			}
+		}
+
+		// Setup contact page and add a form and/or business info
+		$contact_page = '';
+
+		if ( $data['addContactForm'] ) {
+			if ( Jetpack::is_module_active( 'contact-form' ) ) {
+				$contact_page = '[contact-form][contact-field label="Name" type="name" required="true" /][contact-field label="Email" type="email" required="true" /][contact-field label="Website" type="url" /][contact-field label="Message" type="textarea" /][/contact-form]';
```

Comment:
Does this need to be expanded a bit so that the labels can be translated?

Reply:
Good catch, it's updated now. Thanks.